### PR TITLE
LOOP plugins: Solana Relayer + Median

### PR DESCRIPTION
**Pull request by jmank88**
https://smartcontract-it.atlassian.net/browse/BCF-2020

This PR adds experimental support for running the Solana Relayer and Median ReportingPlugin as LOOP Plugins.
This is opt-in only, and not yet supported in the standard docker image (additional binaries required: see `plugins/chainlink.Dockerfile`).

### CODE REVIEW

File: core/services/ocr2/delegate.go [hunk 1]
```diff
@@ -75,12 +84,12 @@ func NewDelegate(
...
 	chainSet evm.ChainSet,
 	lggr logger.Logger,
-	cfg validate.Config,
+	cfg Config,
 	ks keystore.OCR2,
 	dkgSignKs keystore.DKGSign,
 	dkgEncryptKs keystore.DKGEncrypt,
 	ethKs keystore.Eth,
-	relayers map[relay.Network]types.Relayer,
+	relayers map[relay.Network]func() (loop.Relayer, error),
```
Comment (krehermann) [inline core/services/ocr2/delegate.go:92]: this signature is changing, but i don't see any changed calls to NewDelegate. where is this func called? confused way the code is complying...

Comment (jmank88) [reply core/services/ocr2/delegate.go:92]: From `application.go`. The line that makes the call didn't need to change.

Comment (krehermann) [reply core/services/ocr2/delegate.go:92]: i must be missing something. how is it possible change the type in the map without changing changing the caller to use that new type?

File: core/services/ocr2/delegate.go [hunk 2]
```diff
@@ -75,12 +84,12 @@ func NewDelegate(
...
 	mailMon *utils.MailboxMonitor,
 ) *Delegate {
 	return &Delegate{
```
Comment (krehermann) [inline core/services/ocr2/delegate.go:95]: while you are in this area, would you add the field names so we have explicit assignments here?

**Review by krehermann — APPROVED** [review]

**Comment by jmank88** [comment]
Removed a leftover `todo` and rebased.

**Review by krehermann — APPROVED** [review]

**Pull request merged**